src/read/wasm.rs

```diff
@@ -161,22 +139,29 @@ where
     }
 
     fn symbols(&'file self) -> Self::SymbolIterator {
-        let mut data = find_subsection(self.name, 1).unwrap_or(Bytes(&[]));
-        let length = read_u32(&mut data).unwrap_or(0) as usize;
         WasmSymbolIterator {
             file: self,
-            index: 0,
-            length,
-            data,
+            names: self.names_data.and_then(|names| {
+                let func = wp::NameSectionReader::new(names, 0)
+                    .ok()?
+                    .into_iter()
+                    .filter_map(|name| name.ok())
+                    .find_map(|name| match name {
+                        wp::Name::Function(func) => Some(func),
+                        _ => None,
+                    })?;
```

Comment:
Yuck, but oh well. Maybe wasmparser could add something to make this nicer (the equivalent of `find_subsection`).